### PR TITLE
[ReviewDB] Improve UI & Use new RewviewDB api endpoints

### PR DESCRIPTION
**Pull request by mantikafasi**
- changes timestamps from hh:mm to date (more sane)
- updates oauth url to new one
- updates "review @user" component to new discord one
- checks if user is banned on launch,if banned show a modal
- removes some unneccesary webpack searchs
- also prevents "Enter" button in textbox creating new lines
image
image

### CODE REVIEW

File: src/plugins/reviewDB/components/ReviewComponent.tsx [hunk 1]
```diff
@@ -16,11 +16,11 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
 
-import { Settings } from "@api/Settings";
 import { classes } from "@utils/misc";
 import { LazyComponent } from "@utils/react";
 import { filters, findBulk } from "@webpack";
-import { Alerts, moment, Timestamp, UserStore } from "@webpack/common";
+import { Alerts, Timestamp, UserStore } from "@webpack/common";
+import moment from "moment";
```
Comment (Vendicated) [inline src/plugins/reviewDB/components/ReviewComponent.tsx:23]: NO, INSANE

import from webpack common

File: src/plugins/reviewDB/components/ReviewComponent.tsx [hunk 2]
```diff
@@ -89,7 +85,7 @@ export default LazyComponent(() => {
                 }
             }>
 
-                <div className={contents} style={{ paddingLeft: "0px" }}>
+                <div style={{ paddingLeft: "0px" }}>
```
Comment (Vendicated) [inline src/plugins/reviewDB/components/ReviewComponent.tsx:88]:
```suggestion
                <div>
```

File: src/plugins/reviewDB/components/ReviewComponent.tsx [hunk 3]
```diff
@@ -105,18 +101,13 @@ export default LazyComponent(() => {
                     </span>
                     {review.sender.badges.map(badge => <ReviewBadge {...badge} />)}
 
-                    {
-                        !Settings.plugins.ReviewDB.hideTimestamps && (
```
Comment (Vendicated) [inline src/plugins/reviewDB/components/ReviewComponent.tsx:109]: wait why did you kill the setting

File: src/plugins/reviewDB/components/ReviewsView.tsx [hunk 4]
```diff
@@ -27,7 +27,7 @@ import { addReview, getReviews } from "../Utils/ReviewDBAPI";
 import { authorize, showToast } from "../Utils/Utils";
 import ReviewComponent from "./ReviewComponent";
 
-const Classes = findLazy(m => typeof m.textarea === "string");
+const Classes = findLazy(m => m.inputDefault);
```
Comment (Vendicated) [inline src/plugins/reviewDB/components/ReviewsView.tsx:30]:
```suggestion
const Classes = findByPropsLazy("inputDefault");
```

File: src/plugins/reviewDB/index.tsx [hunk 5]
```diff
@@ -68,7 +69,29 @@ export default definePlugin({
             type: OptionType.BOOLEAN,
             description: "Hide timestamps on reviews",
             default: false,
-        }
+        },
+        website: {
+            type: OptionType.COMPONENT,
+            description: "ReviewDB website",
+            component: () => (
+                <Button onClick={() => {
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:77]: you should use proper links (a) instead of buttons

File: src/plugins/reviewDB/index.tsx [hunk 6]
```diff
@@ -82,7 +105,34 @@ export default definePlugin({
                 if (user.lastReviewID !== 0)
                     showToast("You have new reviews on your profile!");
             }
-            settings.userType = user.type;
+
+            if (user.ban_info) {
+                const endDate = new Date(user.ban_info.banEndDate);
```
Comment (Vendicated) [inline src/plugins/reviewDB/index.tsx:110]: why is it a mix of snake and camel case wtf

Comment (mantikafasi) [reply src/plugins/reviewDB/index.tsx:110]: MISTAKE XPLDO

File: src/plugins/reviewDB/components/ReviewComponent.tsx (hunk 1 above)
Comment (mantikafasi) [reply src/plugins/reviewDB/components/ReviewComponent.tsx:23]: soon

File: src/plugins/reviewDB/index.tsx (hunk 5 above)
Comment (mantikafasi) [reply src/plugins/reviewDB/index.tsx:77]: doesnt work explode
I tried to both add button inside <a> component and add "href" to buttons props

**Comment by mantikafasi** [comment]
MERGE

**Comment by Vendicated** [comment]
tbh i dont even wanna look at the code anymore, it always resparks my PTSD maybe i will just merge

**Comment by Vendicated** [comment]
image

File: src/plugins/reviewDB/components/ReviewComponent.tsx [hunk 7]
```diff
@@ -107,16 +103,14 @@ export default LazyComponent(() => {
 
                     {
                         !Settings.plugins.ReviewDB.hideTimestamps && (
-                            <Timestamp
-                                timestamp={moment(review.timestamp * 1000)}
-                                compact={true}
-                            />
-                        )
+                            <Timestamp timestamp={moment(review.timestamp * 1000)} >
+                                {moment(review.timestamp * 1000).format("DD/MM/YYYY")}
```
Comment (Vendicated) [inline src/plugins/reviewDB/components/ReviewComponent.tsx:107]: don't hardcode timestamp format, americans will be very upset

Comment (mantikafasi) [reply src/plugins/reviewDB/components/ReviewComponent.tsx]: love

**Pull request merged**